revisions-information.md

```diff
@@ -0,0 +1,74 @@
+
+## Why do we have revisions?
+
+Revisions are used to provide better metadata, typically in the case when a build-plan turns out to be incorrect, and we wish to tighten bounds to exclude incompatible versions of dependencies, or when a new version of a package is released and loosening dependencies will allow existing packages to work with it. Strategies for managing revisions and their costs are enumerated at: https://github.com/haskell-infra/hackage-trustees/blob/master/cookbook.md#best-practice-for-managing-meta-data
+
+Revisions can also be used when a package homepage or maintainer or the like changes, to keep information displayed current.
+
+## Why not just keep uploading new versions?
+
+First, uploading a whole bunch of code when only metadata changes leads to an unnecessary growth in versions. Second, often revisions need to be applied not only to the most recent version of a package, but prior versions as well. In particular, if a package at a given version has a bad install plan, then you do not want to let some tool continue to think this is a good plan, even if that package is not the latest version,.
+
+## Where can I see revisions?
+
+Packages with revisions will tell you so on their cabal page. You can always access the revisions of a package by appending `/revisions` to its url, like so: http://hackage.haskell.org/package/lens-3.5.1/revisions/
+
+## How can I download a package without revisions applied?
+
+`cabal get --pristine`
+
+## How are revisions numbered?
+
+Revisions are numbered by natural numbers, starting at zero, and increase in increments of 1.
+
+## What can revisions change?
+
+Revisions are intended to only update metadata used by build tooling, or that is used purely for display purposes. The source of a package is never changed, only specified fields of cabal files. Given the same fixed dependencies and flag configuration, as long as a metadata change hasn't disallowed them, the build product from a compiler should be the same regardless of revision changes.
+
+Specifically, revisions can alter the following:
+
+* Dependency bounds (tightening or loosening)
+* Cabal spec version (only between the interval 1.10 and 1.21)
+* Custom-setup dependencies (adding, removing or altering bounds)
+* Flag defaults
+* Flag type (automatic to manual only)
+* Flag descriptions
+* Dependencies (adding, but only in very special cases, see below).
+* Copyright attribution
+* Maintainer attribution
+* Author attribution
+* Tested-with information
+* Bug report URL
+* Homepage URL
+* Source repositories
+* Package synopsis
+* Package description
+* Package category
+
+Revisions can also add custom-setup stanzas. This is necessary because newer cabal library versions make explicit custom-setup depends, while older versions simply shared dependencies with the package proper.
```

Comment:
It's worse than that actually; older versions simply exposed to Setup.hs whatever happened to be registered in the user pkg db; moreover this conflicts with the "no untracked dependency" mantra underlying cabal's design; a link to  https://www.well-typed.com/blog/2015/07/cabal-setup-deps/ would make sense to provide more context